crates/agentgateway/src/proxy/httpproxy.rs

```diff
@@ -1201,17 +1202,42 @@ pub fn build_service_call(
 	} else {
 		return Err(ProxyError::NoHealthyEndpoints);
 	};
+	
+	let Some(ip) = wl.workload_ips.first() else {
+		return Err(ProxyError::NoHealthyEndpoints);
+	};
+	
+	Ok(SocketAddr::from((*ip, target_port)))
+}
+
+pub fn build_service_call(
+	inputs: &ProxyInputs,
+	backend_policies: BackendPolicies,
+	log: &mut Option<&mut RequestLog>,
+	override_dest: Option<SocketAddr>,
+	svc: &Arc<Service>,
+	port: &u16,
+) -> Result<BackendCall, ProxyError> {
+	let port = *port;
+	
+	// Get the endpoint selection details for logging and transport info
+	let workloads = &inputs.stores.read_discovery().workloads;
+	let (_ep, handle, wl) = svc
+		.endpoints
+		.select_endpoint(workloads, svc.as_ref(), port, override_dest)
+		.ok_or(ProxyError::NoHealthyEndpoints)?;
+
+	// Resolve to the actual socket address
+	let dest = resolve_service_endpoint(inputs, svc, port, override_dest)?;
```

Comment:
This is picking the endpoint twice. this will cause various issues like double counting requests, etc (and performance implications)